src/pint/models/noise_model.py

```diff
@@ -345,12 +345,10 @@ def get_ecorrs(self):
             ecorrs.append(getattr(self, ecorr))
         return ecorrs
 
-    def ecorr_basis_weight_pair(self, toas):
-        """Return a quantization matrix and ECORR weights.
+    def get_basis(self, toas):
```

Comment:
Unfortunately, these functions will appear as methods of any TimingModel that has a noise component, so `get_basis` is a little confusing. Perhaps a mention of "noise" in the method name would help make it more obvious what basis you were talking about?

Reply:
I agree. I'll change this.

Reply:
I'll change `get_basis ` -> `get_noise_basis`  and `get_weights` -> `get_noise_weights`.